### PR TITLE
Reduce System.Runtime.Numerics outerloop test time from ~180s to ~90s

### PR DESCRIPTION
**Pull request by stephentoub**
There are two remaining individual tests each taking ~80s doing lots of calculations (RunDoubleExplicitCastFromLargeBigIntegerTests and RunLargeValueLogTests)... I don't know the importance enough of what those are testing to determine whether it's ok to change them, so I've left them as is, which means it's going to be difficult to significantly reduce the overall running time (which on my machine is now ~90s) much further.  Still, this small change brings the running time on my machine down from ~150 seconds to ~90 seconds.

Changes:
- The ModPow3LargeInt test was generated 10 sets of random values, using BigInteger.ModPow with them, and then comparing the result against another, very inefficient implementation of modpow, resulting in a test taking ~150 seconds (and another one calling it with slightly different inputs taking ~30s).  I simply pre-generated a set of 10 inputs and expected outputs (using the same algorithm the test was previously using), and put those into the test as InlineData.

cc: @mellinoe, @axelheer, @Petermarcu

### CODE REVIEW

**Comment by axelheer** [comment]
Just comparing with pre-generated values makes a lot of sense. I didn't like the idea of testing the "real" implementation using another un-tested implementation anyway.

@speshuric can you give a hint about the other two tests mentioned by @stephentoub ?

File: src/System.Runtime.Numerics/tests/BigInteger/modpow.cs [hunk 1]
```diff
@@ -170,30 +170,35 @@ public static void ModPow2Large1SmallInt_Threshold()
...
+        [InlineData("1596188639947986471148999794547338", "685242191738212089917782567856594513073397739443", "41848166029740752457613562518205823134173790454761036532025758411484449588176128053901271638836032557551179866133091058357374964041641117585422447497779410336188602585660372002644517668041207657383104649333118253", "39246949850380693159338034407642149926180988060650630387722725303281343126585456713282439764667310808891687831648451269002447916277601468727040185218264602698691553232132525542650964722093335105211816394635493987")]
+        [InlineData("-1506852741293695463963822334869441845197951776565891060639754936248401744065969556756496718308248025911048010080232290368562210204958094544173209793990218122", "64905085725614938357105826012272472070910693443851911667721848542473785070747281964799379996923261457185847459711", "2740467233603031668807697475486217767705051", "-1905434239471820365929630558127219204166613")]
```
Comment (bartonjs) [inline src/System.Runtime.Numerics/tests/BigInteger/modpow.cs]: These are some pretty ungainly strings.  Do they show up usefully in the test log?  Or should we pack them behind some struct which can ToString as a reasonable value?

Failing that, maybe just chopping the data?

```C#
[InlineData(
"16152447169612934532253858872860101823992426489763666485380167221632010974596350526903901586786157942589749704137347759619657187784260082814484793173551647870373927196771852",
"81750440863317178198977948957223170296434385667401518194303765341584160931262254891905182291359611229458113456891057547210603179481274334549407149567413965138338569615186383",
"-7196805437625531929619339290119232497987723600413951949478998858079935202962418871178150151495255147489121599998392939293913217145771279946557062800866475934948778",
"5616380469802854111883204613624062553120851412450654399190717359559254034938493684003207209263083893212739965150170342564431867389596229480294680122528026782956958")]
```
(Okay, maybe that's not much better)

**Comment by bartonjs** [comment]
It'd be nice if magic could happen to make the inputs more legible.  As long as what shows up in the test log is useful to identify which case failed, LGTM.

File: src/System.Runtime.Numerics/tests/BigInteger/modpow.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime.Numerics/tests/BigInteger/modpow.cs]: > Do they show up usefully in the test log?

I changed a single digit in one of them to see what it looks like when it fails:
```
System.Numerics.Tests.modpowTest.ModPow3LargeInt(value: "-1980488225727187161599876741331015456909269161086"..., exponent: "50051263260206609050451172764283809286167418846283"..., modulu
  s: "-1031394351020333660563633380634987137272001901984"..., expected: "-4559679593639479333979462122256288215243720737403") [FAIL]
        Assert.Equal() Failure
        Expected: -4559679593639479333979462122256288215243720737403
        Actual:   -78084019301444356054762453584152789691856362900873
        Stack Trace:
           c:\Users\stoub\Source\repos\corefx\src\System.Runtime.Numerics\tests\BigInteger\modpow.cs(195,0): at System.Numerics.Tests.modpowTest.ModPow3LargeInt(String value, String exponent,
   String modulus, String expected)
```
That seems reasonable to me both for compactness and ability to identify which one failed.

Similarly all of them show up reasonably in the results .xml file:
```
      <test name="System.Numerics.Tests.modpowTest.ModPow3LargeInt(value: &quot;16152447169612934532253858872860101823992426489763&quot;..., exponent: &quot;81750440863317178198977948957223170296434385667401&quot;..., modulus: &quot;-7196805437625531929619339290119232497987723600413&quot;..., expected: &quot;56163804698028541118832046136240625531208514124506&quot;...)" type="System.Numerics.Tests.modpowTest" method="ModPow3LargeInt" time="0.0198571" result="Pass">
        <traits>
          <trait name="category" value="outerloop" />
        </traits>
      </test>
```

**Comment by stephentoub** [comment]
Thanks for the review, guys.

**Pull request merged**